homeassistant/components/zwave_js/__init__.py

```diff
@@ -418,12 +443,51 @@ async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
     if not entry.data.get(CONF_INTEGRATION_CREATED_ADDON):
         return
 
+    addon_manager: AddonManager = get_addon_manager(hass)
     try:
-        await hass.components.hassio.async_stop_addon(ADDON_SLUG)
-    except HassioAPIError as err:
-        LOGGER.error("Failed to stop the Z-Wave JS add-on: %s", err)
+        await addon_manager.async_stop_addon()
+    except AddonError as err:
+        LOGGER.error(err)
         return
     try:
-        await hass.components.hassio.async_uninstall_addon(ADDON_SLUG)
-    except HassioAPIError as err:
-        LOGGER.error("Failed to uninstall the Z-Wave JS add-on: %s", err)
+        await addon_manager.async_create_snapshot()
+    except AddonError as err:
+        LOGGER.error(err)
+        return
+    try:
+        await addon_manager.async_uninstall_addon()
+    except AddonError as err:
+        LOGGER.error(err)
+
+
+async def async_ensure_addon_running(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Ensure that Z-Wave JS add-on is installed and running."""
+    addon_manager: AddonManager = get_addon_manager(hass)
+    if addon_manager.task_in_progress():
+        raise ConfigEntryNotReady
+    try:
+        addon_is_installed = await addon_manager.async_is_addon_installed()
+        addon_is_running = await addon_manager.async_is_addon_running()
```

Comment:
Both calls fetch the same add-on info but parse it differently.

Not perse necessary to adjust, just an observation.

Reply:
Yeah, I thought about this. We could fetch the info here and parse it but then we're moving the parsing details outside the manager. I wasn't really happy with either solution, but went with this for now.

Reply:
I guess we could have a helper on the manager that returns a result from given add-on info.

Reply:
I guess we could have a helper on the manager that returns a result from given add-on info.

Reply:
state enum: NOT_INSTALLED, INSTALLING, UPDATING, NOT_RUNNING, RUNNING

Reply:
I'll follow up with this later.